e2e/util/kustomize.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// buildKustomization builds the kustomization and returns YAML bytes
+func buildKustomization(kustomizationPath string) ([]byte, error) {
+	fSys := filesys.MakeFsOnDisk()
+
+	options := krusty.MakeDefaultOptions()
+	k := krusty.MakeKustomizer(options)
+
+	resMap, err := k.Run(fSys, kustomizationPath)
+	if err != nil {
+		return nil, fmt.Errorf("kustomize build failed: %w", err)
+	}
+
+	yaml, err := resMap.AsYaml()
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to YAML: %w", err)
+	}
+
+	return yaml, nil
+}
+
+// createK8sClient creates a dynamic Kubernetes client and REST mapper
+func createK8sClient(kubeconfig string) (dynamic.Interface, meta.RESTMapper, error) {
```

Comment:
This should be ideally in env package.

Consider if we can integrate this with the existing Cluster struct:

```go
type Cluster struct {
    Name       string        `json:"name"`
    Client     client.Client `json:"-"`
    Kubeconfig string        `json:"kubeconfig"`
}

type Env struct {
    Hub        *Cluster `json:"hub"`
    PassiveHub *Cluster `json:"passive-hub,omitempty"`
    C1         *Cluster `json:"c1"`
    C2         *Cluster `json:"c2"`
}
```

While the existing Client is for structured operations. We might want to extend the Cluster struct to include both client types. @nirs wdyt?